main.py

```diff
@@ -0,0 +1,157 @@
+import tempfile
+import logging
+from collections import namedtuple
+from datetime import datetime, timedelta
+
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+CertAttributes = namedtuple('CertAttributes', [
+    'country',
+    'state',
+    'city',
+    'org',
+    'org_name',
+    'common'
+])
+
+
+class Cert():
+    """Cert class helps in the generation of SSL certificates."""
+
+    def __init__(self, attributes=None):
+
+        # Accept the attributes that define the certs.
+        if attributes:
+            self.attributes = attributes
+
+    def _private_key():
+        """Generates a private key."""
+        return rsa.generate_private_key(
+            public_exponent=65537,
+            key_size=2048,
+            backend=default_backend()
+        )
+
+    def _certificate(self, certattrs, ca_cert=False):
+        """Generate a certificate."""
+        if not certattrs:
+            certattrs = self.attributes
+
+        key = self._private_key()
+
+        # Create certificate and sign it.
+        subject = issuer = x509.Name([
+            x509.NameAttribute(NameOID.COUNTRY_NAME, certattrs.country),
+            x509.NameAttribute(
+                NameOID.STATE_OR_PROVINCE_NAME, certattrs.state),
+            x509.NameAttribute(NameOID.LOCALITY_NAME, certattrs.city),
+            x509.NameAttribute(NameOID.ORGANIZATION_NAME, certattrs.org),
+            x509.NameAttribute(
+                NameOID.ORGANIZATIONAL_UNIT_NAME, certattrs.org_name),
+            x509.NameAttribute(NameOID.COMMON_NAME, certattrs.common),
+        ])
+        cert = x509.CertificateBuilder()\
+            .subject_name(subject)\
+            .issuer_name(issuer)\
+            .public_key(key.public_key())\
+            .serial_number(x509.random_serial_number())\
+            .not_valid_before(datetime.utcnow())\
+            .not_valid_after(datetime.utcnow() + timedelta(days=3650))\
+            .add_extension(
+                x509.SubjectAlternativeName([x509.DNSName(certattrs.common)]),
+                critical=False
+            )
+
+        if ca_cert:
+            cert = cert.add_extension(
+                x509.BasicConstraints(
+                    ca=True,
+                    path_length=1
+                ),
+                critical=True
+            )
+        cert = cert.sign(key, hashes.SHA256(), default_backend())
```

Comment:
Signing the certificate with it's own key should probably be an option (`self_sign=True`) since it is not always desirable.

Reply:
I think I made the proper modifications to meet this need. LMK what you think of my implementation.